### PR TITLE
Improve Jenkins plugin troubleshooting instructions

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
### What does this PR do? What is the motivation?

Improves Datadog Jenkins plugin troubleshooting instructions.
Some steps are added to troubleshoot missing pipelines data or logs data.

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 1]
```diff
@@ -1455,27 +1455,29 @@ Send pipeline traces.
 ...
 {{< /code-block >}}
 
-### The Datadog Plugin cannot write payloads to the server
+### Pipeline executions data is not available in Datadog
+
+#### HTTP connectivity check
+
+If your Jenkins instance is behind an HTTP proxy, go to **Manage Jenkins** > **Manage Plugins** > **Advanced tab** and make sure the proxy configuration is correct:
+- if the Datadog plugin is configured to send data to Datadog Agent, check that the agent host has been added to the `No Proxy Hosts` section
```
Comment (buraizu) [inline content/en/continuous_integration/pipelines/jenkins.md:1463]:
```suggestion
- If the Datadog plugin is configured to send data to a Datadog Agent, check that the Agent host has been added to the `No Proxy Hosts` section.
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 2]
```diff
@@ -1455,27 +1455,29 @@ Send pipeline traces.
...
+- if the Datadog plugin is configured to send data directly to Datadog ("agentless mode"), check that Datadog host (`datadoghq.com` or `datadoghq.eu`, depending on which Datadog site you're using) has been added to the `No Proxy Hosts` section.
```
Comment (buraizu) [inline content/en/continuous_integration/pipelines/jenkins.md:1464]:
```suggestion
- if the Datadog plugin is configured to send data directly to Datadog ("Agentless mode"), check the Datadog host (`datadoghq.com` or `datadoghq.eu`, depending on which Datadog site you're using) has been added to the `No Proxy Hosts` section.
```

Comment (buraizu) [reply content/en/continuous_integration/pipelines/jenkins.md:1464]: Is this only available for `US1` and `EU` sites?

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 3]
```diff
@@ -1455,27 +1455,29 @@ Send pipeline traces.
...
 
 {{< code-block lang="text" >}}
 Error writing to server
 {{< /code-block >}}
 
-1. If you are using `localhost` as the hostname, try to change it to the server hostname instead.
-2. If your Jenkins instance is behind an HTTP proxy, go to **Manage Jenkins** > **Manage Plugins** > **Advanced tab** and make sure the proxy configuration is correct.
-
-#### HTTP 504
+If you are using `localhost` as the hostname, try to change it to the server hostname instead.
```
Comment (buraizu) [inline content/en/continuous_integration/pipelines/jenkins.md:1474]:
```suggestion
If you are using `localhost` as the hostname, change it to the server hostname instead.
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 4]
```diff
@@ -1455,27 +1455,29 @@ Send pipeline traces.
...
-If the HTTP 504 error message appears, make sure that the Jenkins proxy configuration is correct.
-
-{{< code-block lang="text" >}}
-Failed to send HTTP request: PUT http://localhost:8126/v0.3/traces - Status: HTTP 504
-{{< /code-block >}}
+### Jenkins logs are not available in Datadog
 
-1. If your Jenkins instance is behind an HTTP proxy, go to **Manage Jenkins** > **Manage Plugins** > **Advanced tab** and make sure the proxy configuration is correct.
-  1. Check that `localhost` has been configured in the `No Proxy Hosts` section.
+If the Datadog plugin is configured to send data to Datadog Agent, do the following:
```
Comment (buraizu) [inline content/en/continuous_integration/pipelines/jenkins.md:1478]:
```suggestion
If the Datadog plugin is configured to send data to a Datadog Agent, do the following:
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 5]
```diff
@@ -1455,27 +1455,29 @@ Send pipeline traces.
...
+- make sure that custom log collection over TCP is [enabled and configured][29] in the Agent
+- go to the plugin configuration UI and click "Test logs connection" to verify logs connectivity.
```
Comment (buraizu) [inline content/en/continuous_integration/pipelines/jenkins.md:1480]:
```suggestion
- Make sure that custom log collection over TCP is [enabled and configured][29] in the Agent.
- Go to the plugin configuration UI and click "Test logs connection" to verify logs connectivity.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, just requesting some minor updates and posted a question

File: content/en/continuous_integration/pipelines/jenkins.md (hunk 2 above)
Comment (nikita-tkachenko-datadog) [reply content/en/continuous_integration/pipelines/jenkins.md]: It's available for every site except DD Gov.
I didn't want to list every site explicitly to not be too verbose. All the US sites and the AP one are subdomains of `datadoghq.com` so I assumed including only the top-level domain would be enough.

**Review by buraizu — CHANGES_REQUESTED** [review]

File: content/en/continuous_integration/pipelines/jenkins.md (hunk 2 above)
Comment (buraizu) [reply content/en/continuous_integration/pipelines/jenkins.md]: Thanks for clarifying. From past experience, this kind of omission can cause confusion and lead to support tickets. As such, I've requested to update this language and add a table of the supported sites and host values. I'm happy to keep iterating on this if you think the information could be presented more effectively.

Comment (nikita-tkachenko-datadog) [reply content/en/continuous_integration/pipelines/jenkins.md]: That makes sense, I agree that it's best to make things more explicit.

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 6]
```diff
@@ -1455,27 +1455,37 @@ Send pipeline traces.
...
 
-If the following error message appears in the **Jenkins Log**, make sure that the plugin configuration is correct.
+#### HTTP connectivity check
 
-{{< code-block lang="text" >}}
-Error writing to server
-{{< /code-block >}}
+If your Jenkins instance is behind an HTTP proxy, go to **Manage Jenkins** > **Manage Plugins** > **Advanced tab** and make sure the proxy configuration is correct:
+- If the Datadog plugin is configured to send data to a Datadog Agent, check that the Agent host has been added to the `No Proxy Hosts` section.
+- if the Datadog plugin is configured to send data directly to Datadog ("Agentless mode"), check that Datadog host has been added to the `No Proxy Hosts` section. The table below shows the supported Datadog sites and their corresponding host values:
```
Comment (buraizu) [inline content/en/continuous_integration/pipelines/jenkins.md:1464]:
```suggestion
- If the Datadog plugin is configured to send data directly to Datadog (Agentless mode), check that Datadog host has been added to the `No Proxy Hosts` section. The table below shows the supported Datadog sites and their corresponding host values:
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 7]
```diff
@@ -1455,27 +1455,37 @@ Send pipeline traces.
...
 
-1. If your Jenkins instance is behind an HTTP proxy, go to **Manage Jenkins** > **Manage Plugins** > **Advanced tab** and make sure the proxy configuration is correct.
-  1. Check that `localhost` has been configured in the `No Proxy Hosts` section.
+If you are using `localhost` as the hostname, change it to the server hostname instead.
+
+### Jenkins logs are not available in Datadog
+
+If the Datadog plugin is configured to send data to a Datadog Agent, do the following:
+- Make sure that custom log collection over TCP is [enabled and configured][29] in the Agent.
+- Go to the plugin configuration UI and click "Test logs connection" to verify logs connectivity.
```
Comment (buraizu) [inline content/en/continuous_integration/pipelines/jenkins.md:1488]:
```suggestion
- Go to the plugin configuration UI and click **Test logs connection** to verify logs connectivity.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a couple of minor formatting updates suggested

**Pull request merged**